### PR TITLE
fix `We need to process imported via meta[rel="import"] html` (close #1161)

### LINKED ISSUE #1161: We need to process imported via meta[rel="import"] html
see https://github.com/DevExpress/testcafe/issues/1502

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @miherlosev @churkin

### CODE REVIEW

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

File: src/processing/resources/page.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ class PageProcessor extends ResourceProcessorBase {
             });
         }
 
-        for (var i = result.length; i--; i > -1)
+        for (var i = result.length; i--;)
```
Comment (miherlosev) [inline src/processing/resources/page.js]: Use the standard full  spelling:
```js
for (start;  condition;   step )

for (var i = result.length - 1; i >= 0; i--)
```

File: src/request-pipeline/context.js [hunk 2]
```diff
@@ -51,6 +51,7 @@ export default class RequestPipelineContext {
                 isForm:        parsedResourceType.isForm,
                 isScript:      parsedResourceType.isScript,
                 isEventSource: parsedResourceType.isEventSource,
+                isImport:      parsedResourceType.isImport,
```
Comment (miherlosev) [inline src/request-pipeline/context.js]: `isHTMLImport`

File: src/utils/url.js [hunk 3]
```diff
@@ -24,29 +24,32 @@ export function parseResourceType (resourceType) {
             isIframe:      false,
             isForm:        false,
             isScript:      false,
-            isEventSource: false
+            isEventSource: false,
+            isImport:      false
```
Comment (miherlosev) [inline src/utils/url.js]: `isHtmlImport`

**Comment by LavrovArtem** [comment]
FPR

File: test/server/proxy-test.js [hunk 4]
```diff
@@ -120,6 +120,16 @@ describe('Proxy', function () {
             res.end(fs.readFileSync('test/server/data/page/src.html').toString());
         });
 
+        app.get('/import-page', function (req, res) {
```
Comment (miherlosev) [inline test/server/proxy-test.js]: `app.get('/html-import-page'`...

File: test/server/proxy-test.js [hunk 5]
```diff
@@ -120,6 +120,16 @@ describe('Proxy', function () {
...
+            res.set('content-type', 'text/html');
+            res.end(fs.readFileSync('test/server/data/import-page/src.html').toString());
+        });
+
+        app.get('/import-page-in-iframe', function (req, res) {
```
Comment (miherlosev) [inline test/server/proxy-test.js]: ` app.get('/html-import-page-in-iframe'

File: test/server/proxy-test.js [hunk 6]
```diff
@@ -835,6 +845,63 @@ describe('Proxy', function () {
             });
         });
 
+        it('Should process import pages', function (done) {
```
Comment (miherlosev) [inline test/server/proxy-test.js]: `Should process html import pages`

File: test/server/proxy-test.js [hunk 7]
```diff
@@ -835,6 +845,63 @@ describe('Proxy', function () {
...
+
+            request(options, function (err, res, body) {
+                var expected = fs.readFileSync('test/server/data/import-page/expected.html').toString();
+
+                compareCode(body, expected);
+                done();
+            });
+        });
+
+        it('Should process import pages in iframe', function (done) {
```
Comment (miherlosev) [inline test/server/proxy-test.js]: `'Should process html import pages in iframe`

**Comment by LavrovArtem** [comment]
FPR

**Comment by LavrovArtem** [comment]
ping @churkin

**Pull request merged**